src/modules/privacy-idea-administration/privacy-idea-administration-event-handler.ts

```diff
@@ -19,7 +20,19 @@ export class PrivacyIdeaAdministrationEventHandler {
             event.referrer,
         );
         if (userTokens.length > 0) {
-            await this.privacyIdeaAdministrationService.resetToken(event.referrer);
+            try {
+                await this.privacyIdeaAdministrationService.resetToken(event.referrer);
+                this.logger.info(
+                    `System hat für Benutzer  ${event.referrer} (BenutzerId: ${event.personId}) den 2FA Token zurückgesetzt.`,
+                );
+            } catch (error) {
+                if (error instanceof TokenResetError) {
+                    this.logger.error(
+                        `System hat versucht den 2FA Token von Benutzer ${event.referrer} (BenutzerId: ${event.personId}) zurückzusetzen. Fehler: ${error.message}`,
+                    );
+                    throw error;
+                }
+            }
```

Comment:
If you want to keep the check for the instance, at least move the `throw` outside the if. 
Otherwise errors that aren't instances of `TokenResetError` will silently be ignored.

Maybe remove the if-condition and just pass the error as a second parameter to the logger (If that's enough)

Reply:
done